generator/lookml.py

```diff
@@ -0,0 +1,168 @@
+                attributes["timeframes"] = (
+                    "[\n" + indent(",\n".join(timeframes), " " * 2) + "\n]"
+                )
+            if len(self.path) > 1:
+                attributes["group_label"] = json.dumps(
+                    " ".join(self.path[:-1]).replace("_", " ").title()
+                )
+                attributes["group_item_label"] = json.dumps(
+                    self.path[-1].replace("_", " ").title()
+                )
+            if self.path in MAP_LAYER_NAMES:
+                attributes["map_layer_name"] = MAP_LAYER_NAMES[self.path]
+        return (
+            f"{key}: {name} {{"
+            + indent(
+                "".join(
+                    f"\n{attr}: {value}" for attr, value in sorted(attributes.items())
+                ),
+                " " * 2,
+            )
+            + "\n}"
+        )
+
+
+def _get_dimensions(schema: List[bigquery.SchemaField], *prefix: str):
+    for field in schema:
+        if field.field_type == "RECORD" and not field.mode == "REPEATED":
+            yield from _get_dimensions(field.fields, *prefix, field.name)
+        else:
+            yield Dimension((*prefix, field.name), field.field_type, field.mode)
+
+
+@click.command(help=__doc__)
+@click.option(
+    "--namespaces",
+    default="namespaces.yaml",
+    type=click.File(),
+    help="Path to a yaml namespaces file",
+)
+@click.option(
+    "--target-dir",
+    default="looker-hub/",
+    type=click.Path(),
+    help="Path to a directory where lookml will be written",
+)
+def lookml(namespaces, target_dir):
+    """Generate lookml from namespaces."""
+    client = bigquery.Client()
+    _namespaces = yaml.safe_load(namespaces)
+    target = Path(target_dir)
+    for key, value in _namespaces.items():
+        view_dir = target / key / "views"
+        view_dir.mkdir(parents=True, exist_ok=True)
+        for view, tables in value.get("views", {}).items():
+            path = view_dir / (view + ".view.lkml")
+            # only view one table, either channel="release" or the first one
+            for item in tables:
+                if item.get("channel") == "release":
+                    table = item["table"]
+                    break
+            else:
+                # default to the first table
+                table = tables[0]["table"]
+            dimensions = list(_get_dimensions(client.get_table(table).schema))
+            view_attributes = [
+                indent(f"\n\n{dimension.to_lookml()}", " " * 2)
+                for dimension in dimensions
+            ]
+            for dimension in dimensions:
+                if dimension.path in MEASURE_DIMENSIONS:
+                    view_attributes.append(
+                        "\n"
+                        f"\n  measure: {MEASURE_DIMENSIONS[dimension.path]} {{"
+                        f"\n    sql: COUNT(DISTINCT {'.'.join(dimension.path)}) ;;"
+                        "\n    type: number"
+                        "\n  }"
```

Comment:
This can use the [`type: count_distinct`](https://docs.looker.com/reference/field-reference/measure-type-reference#count_distinct) format

Reply:
added